wpimath/src/main/native/cpp/trajectory/Trajectory.cpp

```diff
@@ -69,44 +71,27 @@ Trajectory::State Trajectory::Sample(units::second_t t) const {
   if (t <= m_states.front().t) return m_states.front();
   if (t >= m_totalTime) return m_states.back();
 
-  // To get the element that we want, we will use a binary search algorithm
-  // instead of iterating over a for-loop. A binary search is O(std::log(n))
-  // whereas searching using a loop is O(n).
-
-  // This starts at 1 because we use the previous state later on for
-  // interpolation.
-  int low = 1;
-  int high = m_states.size() - 1;
-
-  while (low != high) {
-    int mid = (low + high) / 2;
-    if (m_states[mid].t < t) {
-      // This index and everything under it are less than the requested
-      // timestamp. Therefore, we can discard them.
-      low = mid + 1;
-    } else {
-      // t is at least as large as the element at this index. This means that
-      // anything after it cannot be what we are looking for.
-      high = mid;
-    }
-  }
+  // Use binary search to get the element with a timestamp no less than the
+  // requested timestamp. This starts at 1 because we use the previous state
+  // later on for interpolation.
+  const auto& sample =
```

Comment:
std::lower_bound returns an iterator value, so is this creating a reference to a temporary?  Can probably just use "auto" here.

Reply:
This wasn't actually resolved?

Reply:
oh oops, did the wrong one